### ISSUE TITLE
Is DotNetDlib could be used on ARM Platform(Win IOT)?

### ISSUE THREAD

**Issue by zhuxb711** [takuya-takeuchi/Demo#10]
DotNetDlib & DotNetDlib.Native.dll works well in UWP X86. But I want to use it on ARM, So I compile an ARM Version "DotNetDlib.Native.dll" and UWP. However, when I Debug UWP in Win IOT, VS told me that could not found DotNetDlib.Native.dll. 
Acutally,VS could found that .dll because if I use X86- DotNetDlib.Native.dll ,VS will notice me that the architecture of .dll isn't correct.
Setting is correct, DotNetDlib.Native.dll is used as Context(in Resource)
In short, I think VS could found it but maybe  DotNetDlib.Native.dll-ARM could not be used in IOT? Or would you share your idea with me ?How to make it work on Win IOT?
Thanks Sincerely

**Comment by takuya-takeuchi** [comment]
now, i just created library by arm.
but i have not arm machines which are available now.
so please you give me time to answer.

**Comment by takuya-takeuchi** [comment]
I think you have to install VC++ Runtime for ARM.
But there is no suitable runtime for VS 2015 or 2017.
There is VC++ 2013 Runtime for ARM. However it is allowed to install for only MS surface RT.

Windows 10 IoT Core can use UWP application and UWP use system native library.
Maybe we are not allowed to use no system native library.

**Comment by zhuxb711** [comment]
In VisualStudio-Project-Add Reference, I found something like VC++Runtime Reference, but it seems useless.

**Comment by zhuxb711** [comment]
I think a better solution is compile Dlib by using Windows Runtime Components. The OpenCV could work well on IOT by this way.

**Comment by takuya-takeuchi** [comment]
Ok. But as you know, dlib does not say it can work at Windows IoT.
So Even though I imitate opencv's approach, it cannot guarantee to get it work.

However, it us interesting your advice.
I want to try it.

**Comment by takuya-takeuchi** [comment]
CMAKE_SYSTEM_NAME looks like to be good.
https://cmake.org/cmake/help/v3.4/manual/cmake-toolchains.7.html#id11

**Comment by takuya-takeuchi** [comment]
**NOTE**
- dlib defined ***WIN32*** preprocessor if use ***_MSC_VER*** preprocessor. But it occurs compile error due to missing ***CoCreateInstanceFromApp***.
- ***timeGetTime*** is not provide in UWP. ***timestamper::get_timestamp*** uses it in ***misc_api_kernel_1.cpp***.
  - Comment out and return 0 always. But dlib uses timestamper::get_timestamp. So this modification occurs some issue.
  - ***QueryPerformanceCounter*** can replate it? But some hardware may not support QueryPerformanceCounter.

**Comment by takuya-takeuchi** [comment]
Yes, you are correct. I hope that dlib supports UWP.
Fow now, I modified dlib source and CMakeLists.txt, I could work UWP sample with DlibDotNet.

![uwp](https://user-images.githubusercontent.com/6241854/43974767-8005bb9e-9d16-11e8-93c0-b1255379c7c5.gif)

**Comment by zhuxb711** [comment]
Yeap, Dlib could work well on UWP-X86 as you saw. It just don't work on UWP-ARM. I hope one day Dlib could work on UWP-ARM, I'm very appreciate for your answer. And I will stay focus on your Project. Thanks for your passion on making the project better on UWP.

**Comment by zhuxb711** [comment]
Brilliant！I will follow you step later. Besides, I want to know why Native.dll work by adding Dnn.dll. And what the Dnn.dll used for?

**Comment by takuya-takeuchi** [comment]
> I want to know why Native.dll work by adding Dnn.dll

At first, I did not have any intention to add it to project.
Because it is one binary which is consist of DliDotNet package.

However, I removed Dnn.dll from project and uploadd new package, it did NOT work.

Though, any functions of this dll should not be used for above sample project.
It is weird.

So I debugged sample project via Visual Studio Remote Debugger but it worked fine.
![default](https://user-images.githubusercontent.com/6241854/44204677-99a5ad80-a18e-11e8-9bf0-b79b940f99ff.png)

There is any difference between debug mode and release package, right?

> And what the Dnn.dll used for?

Dnn.dll is used for using deep neural network of dlib.
Generally, DNN requires high performance machine or GPU because DNN task is very heavy.
This sample project uses the following api

- Dlib.GetFrontialFaceDetector
- Matrix
- Dlib.LoadImage
- FrontialFaceDetector.Operator

But above interface should depend on DlibDotNet.Native.dll.

**Comment by zhuxb711** [comment]
So, Native.dll needs Dnn.dll for unknown reasons? Obversely, no function use Dnn.dll. That's strange.

**Comment by takuya-takeuchi** [comment]
The following is command line which I use when comple

````
set DLIBVER=19.15
cmake -G "Visual Studio 15 2017 ARM" -T host=x64 ^
           -DDLIB_PATH=D:/Works/Lib/DLib/%DLIBVER% ^
           -DCMAKE_SYSTEM_NAME=WindowsStore ^
           -DCMAKE_SYSTEM_VERSION=10.0 ^
           -DWINAPI_FAMILY=WINAPI_FAMILY_APP ^
           -D_WINDLL=ON ^
           -D_WIN32_UNIVERSAL_APP=ON ^
           -DDLIB_HAVE_SSE2=OFF ^
````

You should use ***-T host=x64** to use x64 tool chain.
It does not mean compiler generate x64 binary.
x64 tool chain is x64 binary and it can treat huge heap memory.

**Comment by zhuxb711** [comment]
![image](https://user-images.githubusercontent.com/42038175/44247897-20539c80-a21a-11e8-9743-ebd2cb4d6009.png)
Native.dll as Context and AlwaysCopy
Native.Dnn.dll as Context and AlwaysCopy
![image](https://user-images.githubusercontent.com/42038175/44247959-71639080-a21a-11e8-8ac1-1248fd33fe3b.png)
But it doesn't work.
"Exception thrown: 'System.DllNotFoundException' in DlibDotNet.dll"
So DlibDotNet.dll seems could not found Native.dll and Native.Dnn.dll
Dumpbin confirmed Native.dll and Native.Dnn.dll is ARM, DlibDotNet.dll is X86
I don't know how to do next.Please tell me what's wrong.
Thanks.

**Comment by takuya-takeuchi** [comment]
It seems like that you packages fails to include DlibDotNet.Native.dll and Dnn.dll.
Could you upload attached file to you windows iot core after extracting?
(It is only zipped ***FaceDetection_1.0.0.0_arm.appxbundle***)

[FaceDetection_1.0.0.0_arm.zip](https://github.com/takuya-takeuchi/Demo/files/2296127/FaceDetection_1.0.0.0_arm.zip)

And you can compare between it and your package, then you can see something.

**Comment by zhuxb711** [comment]
Your FaceDetection works well on IOT. Could you give your SourceCode to me ? My program always told me that it could not found dll....sadness....

**Comment by takuya-takeuchi** [comment]
Please get source from https://github.com/takuya-takeuchi/DlibDotNet/tree/develop/examples/UWP/FaceDetection

This sample is same as above one.
